internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -385,13 +393,29 @@ func (r *ClusterDeploymentsReconciler) installDay2Hosts(ctx context.Context, log
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) createNoIngressKubeConfig(ctx context.Context, log logrus.FieldLogger, cluster *hivev1.ClusterDeployment, c *common.Cluster, clusterInstall *hiveext.AgentClusterInstall) error {
+
+	k, err := r.ensureKubeConfigNoIngressSecret(ctx, log, cluster, c)
```

Comment:
Not sure that we need to create or try to create it on every reconcile, maybe instead you can check if `clusterInstall.Spec.ClusterMetadata` is already defined? 
If user delete the service it's going to be his problem.

Reply:
It will create only once. We return if found:
```
func (r *ClusterDeploymentsReconciler) ensureKubeConfigNoIngressSecret(ctx context.Context, log logrus.FieldLogger, cluster *hivev1.ClusterDeployment, c *common.Cluster) (*corev1.Secret, error) {
	s := &corev1.Secret{}
	name := fmt.Sprintf(adminKubeConfigStringTemplate, cluster.Name)
	getErr := r.Get(ctx, types.NamespacedName{Namespace: cluster.Namespace, Name: name}, s)
	if getErr == nil || !k8serrors.IsNotFound(getErr) {
		return s, getErr
	}
```

Reply:
I mean that you will not need to call for get api at all.

Reply:
Done